src/outputs/parthenon_hdf5.cpp

```diff
@@ -493,14 +505,34 @@ void PHDF5Output::WriteOutputFileImpl(Mesh *pm, ParameterInput *pin, SimTime *tm
       HDF5WriteND(g_var, vname, host_data.data(), vinfo.tensor_rank + 1, local_offset,
                   local_count, global_count, pl_xfer, H5P_DEFAULT);
     }
+    std::vector<Real> pos_tmp; // tmp vector to (potentially) hold particle positions
     auto &rvars = std::get<SwarmInfo::MapToVarVec<Real>>(swinfo.vars);
     for (auto &[vname, swmvarvec] : rvars) {
       const auto &vinfo = swinfo.var_info.at(vname);
       auto host_data = swinfo.FillHostBuffer(vname, swmvarvec);
       SetCounts(swinfo, vinfo);
       HDF5WriteND(g_var, vname, host_data.data(), vinfo.tensor_rank + 1, local_offset,
                   local_count, global_count, pl_xfer, H5P_DEFAULT);
+      if (output_params.write_swarm_xdmf &&
+          (vname == swarm_position::x::name() || vname == swarm_position::y::name() ||
+           vname == swarm_position::z::name())) {
+        pos_tmp.insert(pos_tmp.end(), host_data.begin(), host_data.end());
```

Comment:
Is it necessary to fill this contiguous block like this? I feel like we could probably get around this by using HDF5's lower level memory space machinery to tell it to write swarm::x specifically into the first "column," swarm::y directly into the second column, etc. Maybe not though. Haven't thought about it carefully.

Reply:
I was already shooting in the dark with the existing Parthenon HDF functions.  I think you are among the few that actually understands the nuts and bolts of those.  If you immediately see how to do that, I would welcome that strategy with open arms.  This hack I put in is quite ugly.

Reply:
I don't think the parthenon HDF5 functions can do it. We'd have to go to the lower-level HDF5. Let's talk either tomorrow or next week (depending on when you're around). I can try to suggest a hack.

Reply:
We discussed offline---I think it's possible to do this with low level HDF5 but unclear if this is needed.